src/main/webapp/app/exercises/shared/structured-grading-criterion/grading-instructions-details/grading-instructions-details.component.ts

```diff
@@ -459,4 +485,16 @@ export class GradingInstructionsDetailsComponent implements OnInit, AfterContent
         const criterionIndex = this.exercise.gradingCriteria!.indexOf(criterion);
         this.exercise.gradingCriteria!.splice(criterionIndex, 1);
     }
+
+    /**
+     * @function setExerciseGradingInstructionText
+     * @desc Gets a tuple of text and domainCommandIdentifiers and assigns text values as grading instructions of exercise
+     * @param domainCommands containing tuples of [text, domainCommandIdentifiers]
+     */
+    setExerciseGradingInstructionText(domainCommands: [string, DomainCommand | null][]): void {
```

Comment:
Is there a reason why we don't create a model class for `[string, DomainCommand | null]`? 

Something like:
```typescript
class DomainCommand {
  let text: string;
  let domainId: DomainCommandIdentifier;
}
```

Guess it would add more clarity.

Reply:
do not have special reason. there are several places that this structure is used as an input in funcs related with markdown... agree with you, this form seems a little bit ugly to me. I will make some follow-ups and I'll work on same components. I may refactor it. need to have a look, thanks for suggestion, coool 👍